Sources/NIOCore/ByteBuffer-hexdump.swift

```diff
@@ -0,0 +1,188 @@
+            result += "  "
+
+            // Center column consists of:
+            // - xxd-compatible dump of the first 8 bytes
+            // - space
+            // - xxd-compatible dump of the rest 8 bytes
+            // If there are not enough bytes to dump, the column is padded with space.
+            let left = slice.getSlice(at: 0, length: min( 8, slice.readableBytes ))!
+            let right = slice.getSlice(at: 8, length: min( 8, slice.readableBytes - 8)) ?? ByteBuffer()
+            let leftHex = left.hexDumpShort()
+            let rightHex = right.hexDumpShort()
+
+            result += leftHex
+            result += String(repeating: " ", count: 25 - leftHex.count)
+            result += rightHex
+            result += String(repeating: " ", count: 25 - rightHex.count)
+
+            // Right column renders the 16 bytes line as ASCII characters, or "." if the character is not printable.
+            let printableBytes = slice.readableBytesView.map {
+                let printableRange = UInt8(ascii: " ") ..< UInt8(ascii: "~")
+                return printableRange.contains($0) ? $0 : UInt8(ascii: ".")
+            }
+            result += "|\(String(decoding: printableBytes, as: UTF8.self))|\n"
+
+            lineOffset += lineLength
+        }
+        return result
+    }
+
+    /// Returns a `String` of hexadecimal digits of bytes in the Buffer,
+    /// with formatting compatible with output of `hexdump -C`.
+    func hexDumpLong() -> String {
+        var result = self._hexDumpLines(offset: 0)
+        result += String(byte: self.readableBytes, padding: 8)
+        return result
+    }
+
+    /// Returns a `String` of hexadecimal digits of bytes in this ByteBuffer
+    /// with formatting sort of compatible with `hexdump -C`, but clipped on length.
+    /// Dumps limit/2 first and limit/2 last bytes, with a separator line in between.
+    ///
+    /// - parameters:
+    ///     - limit: Max bytes to dump.
+    func hexDumpLong(limit: Int) -> String {
+        guard readableBytes > limit else {
+            return self.hexDumpLong()
+        }
+
+        let separator = "                                     ...                                      \n"
+        var result = ""
+
+        var buffer = self
+        let front = buffer.readSlice(length: limit / 2)!
+        buffer.moveReaderIndex(forwardBy: buffer.readableBytes - limit / 2)
+        let back = buffer.readSlice(length: buffer.readableBytes)!
+
+        result += front._hexDumpLines(offset: 0)
+        result += separator
+        result += back._hexDumpLines(offset: self.readableBytes - limit/2)
+        result += String(byte: self.readableBytes, padding: 8)
+        return result
+    }
+
+    /// hexDump this `ByteBuffer` in a preferred `HexDumpFormat`.
+    /// hexDump will dump only the readable bytes, from `readerIndex` to `writerIndex`. You're responsible for setting those indices or slicing into the part that you want to be dumped.
+    ///
```

Comment:
I think we can skip the part about the reader and writer index; we generally advise users to not move them manually but to have them moved by using the various `read/`write` methods.

```suggestion
    /// Returns a hex dump of the readable bytes of this `ByteBuffer` in a preferred `HexDumpFormat`.
```